src/main/java/it/gov/pagopa/pu/organization/service/BrokerEncryptionService.java

```diff
@@ -0,0 +1,48 @@
+package it.gov.pagopa.pu.organization.service;
+
+import it.gov.pagopa.pu.organization.dto.generated.BrokerApiKeys;
+import it.gov.pagopa.pu.organization.model.Broker;
+import it.gov.pagopa.pu.organization.util.AESUtils;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+@Service
+@Slf4j
+public class BrokerEncryptionService {
+
+  private final String brokerEncryptPassword;
+
+  BrokerEncryptionService(
+    @Value("${app.brokerEncryptPassword}") String brokerEncryptPassword) {
+    this.brokerEncryptPassword = brokerEncryptPassword;
+  }
+
+  private final Map<byte[], String> apiKeyDecryptMap = Collections.synchronizedMap(new HashMap<>());
+
+  public BrokerApiKeys getBrokerDecryptedApiKeys(Broker broker){
+    return BrokerApiKeys.builder()
+      .syncKey(decryptKey(broker.getSyncKey(),"SYNC", broker.getBrokerId()))
+      .acaKey(decryptKey(broker.getAcaKey(),"ACA", broker.getBrokerId()))
+      .gpdKey(decryptKey(broker.getGpdKey(),"GPD", broker.getBrokerId()))
+      .build();
+  }
+
+  private String decryptKey(byte[] encryptedKey, String type, Long brokerId){
+    if(encryptedKey==null || encryptedKey.length==0) {
+      log.debug("null or empty api-key");
+      return null;
+    }
+    String decrypted = apiKeyDecryptMap.computeIfAbsent(encryptedKey, c -> {
+      log.debug("invoking AESUtils to decrypt api-key");
+      return AESUtils.decrypt(brokerEncryptPassword,c);
+    });
+    log.debug("decrypted api-key[{}] for broker[{}]: {}", type, brokerId, StringUtils.abbreviateMiddle(decrypted, "..", 8));
+    return decrypted;
```

Comment:
Even in debug level we should avoid to print secrets or PII.
```suggestion
    return apiKeyDecryptMap.computeIfAbsent(encryptedKey, c -> {
      log.debug("invoking AESUtils to decrypt {} api-key of brokerId {}", type, brokerId);
      return AESUtils.decrypt(brokerEncryptPassword,c);
    });
```

Reply:
not sure about this... It may be really useful for debug reasons to verify what is the Api-Key used for invaction on ACA (or GPD) call, to understand if the right value has been used, and it is quite difficult to verify it otherwise.
And printing only the first/last 3 characters preserves the confidentiality of keys.